### PR TITLE
feat: Adds the ability to validate password at signup

### LINKED ISSUE #4209: Captcha gets invalidated after submission
- Password validation is not correct
- If Password doesn't have special characters, submission would not work. 
- Captcha would expire after submission. Chatwoot don't refresh the captcha
- Subsequent submission would fail as the captcha is not valid. 
- Chatwoot doesn't show the valid error if captcha is invalid.

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes #4209 & https://github.com/chatwoot/product/issues/398

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Validation**

`Before`

1. Minimum length for a password is 6.

`After`

1. Minimum length for a password is 6.
2. Must include one capital letter.
3. Must include one small letter.
4. Should include one numerical value between `0-9`.
5. Must include one special character like `[!@#$%^&*()_+\-=[\]{}|'"/\\.,``<>:;?~]`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/shared/helpers/Validators.js [hunk 1]
```diff
@@ -2,3 +2,12 @@ export const isPhoneE164 = value => !!value.match(/^\+[1-9]\d{1,14}$/);
 export const isPhoneE164OrEmpty = value => isPhoneE164(value) || value === '';
 export const shouldBeUrl = (value = '') =>
   value ? value.startsWith('http') : true;
+export const isValidPassword = value => {
```
Comment (pranavrajs) [inline app/javascript/shared/helpers/Validators.js:5]: @iamsivin Can you please add a test case for this method?

File: app/javascript/shared/helpers/Validators.js [hunk 2]
```diff
@@ -2,3 +2,12 @@ export const isPhoneE164 = value => !!value.match(/^\+[1-9]\d{1,14}$/);
...
+  const containsUppercase = /[A-Z]/.test(value);
+  const containsLowercase = /[a-z]/.test(value);
+  const containsNumber = /[0-9]/.test(value);
+  const containsSpecial = /[%w!@#$%^&*()_+{}|'"^.,`<>:;?=~]/.test(value);
+  return (
+    containsUppercase && containsLowercase && containsNumber && containsSpecial
+  );
```
Comment (pranavrajs) [inline app/javascript/shared/helpers/Validators.js:12]:
```suggestion
  const containsSpecialCharacter = /[%w!@#$%^&*()_+{}|'"^.,`<>:;?=~]/.test(value);
  return (
    containsUppercase && containsLowercase && containsNumber && containsSpecialCharacter
  );
```

File: app/javascript/shared/helpers/Validators.js (hunk 1 above)
Comment (iamsivin) [reply app/javascript/shared/helpers/Validators.js:5]: Sure

File: app/javascript/shared/helpers/specs/ValidatorsHelper.spec.js [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
 
 describe('#shouldBeUrl', () => {
   it('should return correct url', () => {
     expect(shouldBeUrl('http')).toEqual(true);
   });
 });
+
+describe('#isValidPassword', () => {
+  it('should return correct password', () => {
+    expect(isValidPassword('testPass4!')).toEqual(true);
```
Comment (muhsin-k) [inline app/javascript/shared/helpers/specs/ValidatorsHelper.spec.js:12]: Add false conditions too

**Pull request merged**